diff --git a/chocopy-llvm/lib/Parser/Parser.cpp b/chocopy-llvm/lib/Parser/Parser.cpp
index ad1fa237e816077ed2ccd469304093cfb6b67b94..0d368691f35b555468c38e65aa29fbfc8267b8dd 100644
--- a/chocopy-llvm/lib/Parser/Parser.cpp
+++ b/chocopy-llvm/lib/Parser/Parser.cpp
@@ -807,55 +807,50 @@ Expr *Parser::parsePrimaryExpr() {
     return parseLiteral();
 
   emitUnexpected();
   return nullptr;
 }
 
 /// type = ID | IDSTRING | '[' type ']'
 TypeAnnotation *Parser::parseType() {
   SMRange Loc = Tok.getLocation();
   switch (Tok.getKind()) {
   case tok::identifier: {
     StringRef Name = Tok.getSymbolInfo()->getName();
     consumeToken();
     return Context.createClassType(Loc, Name);
   }
   case tok::idstring: {
     StringRef Name = Tok.getLiteralData();
     consumeToken();
     return Context.createClassType(Loc, Name);
   }
   case tok::l_square: {
     consumeToken();
     if (TypeAnnotation *T = parseType()) {
       if (expect(tok::r_square)) {
         SMLoc End = Tok.getLocation().End;
-        const char *Ptr = End.getPointer();
-        if (Ptr && *Ptr == ' ')
-          End = SMLoc::getFromPointer(Ptr + 2);
-        else
-          End = SMLoc::getFromPointer(Ptr + 1);
         consumeToken();
         Loc = SMRange(Loc.Start, End);
         return Context.createListType(Loc, T);
       }
     }
     return nullptr;
   }
   default:
     emitUnexpected();
     return nullptr;
   }
 }
 
 Literal *Parser::parseLiteral() {
   SMRange Loc = Tok.getLocation();
 
   if (consumeToken(tok::kw_None)) {
     return Context.createNoneLiteral(Loc);
   } else if (consumeToken(tok::kw_True)) {
     return Context.createBooleanLiteral(Loc, true);
   } else if (consumeToken(tok::kw_False)) {
     return Context.createBooleanLiteral(Loc, false);
   } else if (Tok.is(tok::integer_literal)) {
     llvm::APInt Value(32, Tok.getLiteralData(), 10);
     consumeToken();
