diff --git a/chocopy-llvm/include/chocopy-llvm/Lexer/Lexer.h b/chocopy-llvm/include/chocopy-llvm/Lexer/Lexer.h
index df53fe0c0bbfe5e5054a4703a2528f1e73b945da..9a1393d7973040efd11caaef58071b4e0bd86b51 100644
--- a/chocopy-llvm/include/chocopy-llvm/Lexer/Lexer.h
+++ b/chocopy-llvm/include/chocopy-llvm/Lexer/Lexer.h
@@ -1,43 +1,43 @@
 #ifndef CHOCOPY_LLVM_LEXER_LEXER_H
 #define CHOCOPY_LLVM_LEXER_LEXER_H
 
 #include "chocopy-llvm/Basic/SymbolTable.h"
 #include "chocopy-llvm/Lexer/Token.h"
 
 #include <llvm/Support/SourceMgr.h>
 
 namespace chocopy {
 class DiagnosticsEngine;
 
 class Lexer {
 public:
   Lexer(DiagnosticsEngine &Diags, llvm::SourceMgr &SrcMgr);
 
   DiagnosticsEngine &getDiagnostics() const { return Diags; }
 
-  SymbolTable &getSymbolTable() { return SymbolTable; }
+  SymbolTable &getSymbolTable() { return SymTable; }
 
   void reset();
 
   /// Lex returns true if function returns Tok
   bool lex(Token &Tok);
 
   const Token &LookAhead(unsigned N) {
     if (CachedTokenPos + N < CachedTokens.size())
       return CachedTokens[CachedTokenPos + N];
     else
       return PeekAhead(N + 1);
   }
 
   /// @todo Add SourceLocation
   /// SourceLocation getLocation();
 private:
   void enterCachingMode();
   void exitCachingMode();
 
   const Token &PeekAhead(unsigned N);
 
   bool lexImpl(Token &Tok);
   void cachingLexImpl(Token &Tok);
 
   bool isEof() const { return BufPtr == BufEnd; }
@@ -52,31 +52,31 @@ private:
   static bool callbackLexer(Lexer &Lexer, Token &Tok) {
     return Lexer.lexImpl(Tok);
   }
 
   static bool callbackCachingLexer(Lexer &Lex, Token &Tok) {
     Lex.cachingLexImpl(Tok);
     return true;
   }
 
 private:
   using CachedTokensTy = SmallVector<Token, 1>;
   using LexerCallback = bool(Lexer &, Token &);
 
   DiagnosticsEngine &Diags;
   llvm::SourceMgr &SourceMgr;
   CachedTokensTy CachedTokens;
   CachedTokensTy::size_type CachedTokenPos = 0;
   SmallVector<int> IndentStack = {0};
   LexerCallback *CurLexerCallback = callbackLexer;
   unsigned CurBuffer = 0;
   StringRef CurBuf;
   const char *IndentPtr = nullptr;
   const char *BufPtr = nullptr;
   const char *BufEnd = nullptr;
   bool IsCachingMode = true;
-  SymbolTable SymbolTable;
+  SymbolTable SymTable;
   int DedentCount = 0;
   bool IsLogLineStart = false;
 };
 } // namespace chocopy
-#endif // CHOCOPY_LLVM_LEXER_LEXER_H
\ No newline at end of file
+#endif // CHOCOPY_LLVM_LEXER_LEXER_H
diff --git a/chocopy-llvm/lib/Lexer/Lexer.cpp b/chocopy-llvm/lib/Lexer/Lexer.cpp
index ed0bdba4dd94885bcfdd8f0a14d0bdc81a0828f0..095dd8e23b5fc21bc72b980a355b12076c61031e 100644
--- a/chocopy-llvm/lib/Lexer/Lexer.cpp
+++ b/chocopy-llvm/lib/Lexer/Lexer.cpp
@@ -8,63 +8,63 @@
 namespace chocopy {
 static llvm::StringMap<tok::TokenKind> KwTable = {
 #define KEYWORD(KW, STR) std::make_pair(StringRef(STR), tok::kw_##KW),
 #include "chocopy-llvm/Lexer/TokenKinds.def"
 };
 
 bool isKeyword(StringRef Identifier) { return KwTable.contains(Identifier); }
 
 tok::TokenKind getKeword(StringRef Identifier,
                          tok::TokenKind DefaultTokenCode = tok::unknown) {
   auto Result = KwTable.find(Identifier);
   if (Result != KwTable.end())
     return Result->second;
   return DefaultTokenCode;
 }
 
 static void initializeSymbolTable(SymbolTable &ST) {
   for (auto &Item : KwTable)
     ST.get(Item.getKey(), Item.getValue());
 }
 
 Lexer::Lexer(DiagnosticsEngine &Diags, llvm::SourceMgr &SrcMgr)
     : Diags(Diags), SourceMgr(SrcMgr), CurBuffer(SourceMgr.getMainFileID()),
       CurBuf(SourceMgr.getMemoryBuffer(CurBuffer)->getBuffer()),
       BufPtr(CurBuf.begin()), BufEnd(CurBuf.end()) {
-  initializeSymbolTable(SymbolTable);
+  initializeSymbolTable(SymTable);
 }
 
 void Lexer::reset() {
   IndentStack = {0};
   DedentCount = 0;
   IsLogLineStart = false;
   BufPtr = CurBuf.begin();
   CurLexerCallback = callbackLexer;
   CachedTokenPos = 0;
   CachedTokens.clear();
-  SymbolTable = chocopy::SymbolTable();
-  initializeSymbolTable(SymbolTable);
+  SymTable = chocopy::SymbolTable();
+  initializeSymbolTable(SymTable);
 }
 
 bool Lexer::lex(Token &Tok) {
   while (!CurLexerCallback(*this, Tok))
     ;
   return true;
 }
 
 bool Lexer::lexImpl(Token &Tok) {
   Tok.startToken();
 
   if (DedentCount) {
     SMLoc L = SMLoc::getFromPointer(IndentPtr);
     Tok.setKind(tok::DEDENT);
     Tok.setDedentData(IndentPtr);
     Tok.setLocation(SMRange(L, L));
     DedentCount--;
     return true;
   }
 
   const char *Ptr = BufPtr;
   while (!IsLogLineStart && !isEof()) {
     int Indention = 0;
     while (!isEof()) {
       if (*BufPtr == ' ')
@@ -184,51 +184,51 @@ bool Lexer::lexImpl(Token &Tok) {
       readNext();
     return true;
   }
 
   assert(isEof());
 
   if (IndentStack.empty()) {
     Tok.setKind(tok::eof);
     Tok.setEofData(BufPtr);
     return true;
   }
 
   DedentCount = IndentStack.size() - 1;
   IndentStack.clear();
   return false;
 }
 
 void Lexer::handleIdentifier(Token &Tok) {
   const char *Ptr = BufPtr;
   while (isChocopyIdentifierBody(*BufPtr))
     readNext();
 
   unsigned Length = BufPtr - Ptr;
   SMLoc B = SMLoc::getFromPointer(Ptr);
   SMLoc E = SMLoc::getFromPointer(BufPtr);
-  SymbolInfo &SI = SymbolTable.get(StringRef(Ptr, Length));
+  SymbolInfo &SI = SymTable.get(StringRef(Ptr, Length));
   Tok.setKind(SI.getKind());
   Tok.setLocation(SMRange(B, E));
   Tok.setSymbolInfo(&SI);
   Tok.setLength(Length);
 }
 
 void Lexer::handleIntegerLiteral(Token &Tok) {
   const char *Ptr = BufPtr;
   while (isDigit(*BufPtr))
     readNext();
 
   unsigned Length = BufPtr - Ptr;
   SMLoc B = SMLoc::getFromPointer(Ptr);
   SMLoc E = SMLoc::getFromPointer(BufPtr);
   Tok.setKind(tok::integer_literal);
   Tok.setLocation(SMRange(B, E));
   Tok.setLiteralData(Ptr);
   Tok.setLength(Length);
 }
 
 void Lexer::handleString(Token &Tok) {
   bool IsId = true;
 
   const char *QuoteStart = BufPtr;
   // Skip opening quote.
@@ -249,36 +249,36 @@ void Lexer::handleString(Token &Tok) {
     readNext();
   }
 
   const char *End = BufPtr;
   unsigned Length = End - Start;
 
   if (IsId && Length > 0 && isChocopyIdentifierHead(*Start))
     Tok.setKind(tok::idstring);
   else
     Tok.setKind(tok::string_literal);
 
   SMLoc B = SMLoc::getFromPointer(QuoteStart);
   const char *LocEnd = End;
   if (!isEof() && *BufPtr == '\"')
     LocEnd = BufPtr + 1;
   SMLoc E = SMLoc::getFromPointer(LocEnd);
   Tok.setLocation(SMRange(B, E));
   Tok.setLiteralData(Start);
   Tok.setLength(Length);
 
   if (!isEof() && *BufPtr == '\"')
     readNext();
 }
 
 void Lexer::handleToken(Token &Tok, StringRef Str, tok::TokenKind Kind) {
-  SymbolInfo &SI = SymbolTable.get(Str, Kind);
+  SymbolInfo &SI = SymTable.get(Str, Kind);
   Tok.setKind(SI.getKind());
   SMLoc B = SMLoc::getFromPointer(BufPtr);
   SMLoc E = SMLoc::getFromPointer(BufPtr + SI.getName().size());
 
   Tok.setLocation(SMRange(B, E));
   Tok.setSymbolInfo(&SI);
   Tok.setLength(SI.getName().size());
   BufPtr += Str.size();
 }
 } // namespace chocopy
