diff --git a/chocopy-llvm/lib/Lexer/Lexer.cpp b/chocopy-llvm/lib/Lexer/Lexer.cpp
index 9b49ec573de92e6d839074328c67ca54f59f2847..ed0bdba4dd94885bcfdd8f0a14d0bdc81a0828f0 100644
--- a/chocopy-llvm/lib/Lexer/Lexer.cpp
+++ b/chocopy-llvm/lib/Lexer/Lexer.cpp
@@ -207,66 +207,78 @@ void Lexer::handleIdentifier(Token &Tok) {
   SMLoc B = SMLoc::getFromPointer(Ptr);
   SMLoc E = SMLoc::getFromPointer(BufPtr);
   SymbolInfo &SI = SymbolTable.get(StringRef(Ptr, Length));
   Tok.setKind(SI.getKind());
   Tok.setLocation(SMRange(B, E));
   Tok.setSymbolInfo(&SI);
   Tok.setLength(Length);
 }
 
 void Lexer::handleIntegerLiteral(Token &Tok) {
   const char *Ptr = BufPtr;
   while (isDigit(*BufPtr))
     readNext();
 
   unsigned Length = BufPtr - Ptr;
   SMLoc B = SMLoc::getFromPointer(Ptr);
   SMLoc E = SMLoc::getFromPointer(BufPtr);
   Tok.setKind(tok::integer_literal);
   Tok.setLocation(SMRange(B, E));
   Tok.setLiteralData(Ptr);
   Tok.setLength(Length);
 }
 
 void Lexer::handleString(Token &Tok) {
   bool IsId = true;
-  const char *Start = readNext();
-  while (!isEof() && *Start != *BufPtr && !isVerticalWhitespace(*BufPtr)) {
+
+  const char *QuoteStart = BufPtr;
+  // Skip opening quote.
+  readNext();
+  const char *Start = BufPtr;
+
+  while (!isEof() && *BufPtr != '\"' && !isVerticalWhitespace(*BufPtr)) {
     if (*BufPtr == '\\') {
       IsId = false;
-      if (++BufPtr != BufEnd)
+      readNext();
+      if (!isEof())
         readNext();
       continue;
     }
 
-    if (!isChocopyIdentifierBody(*readNext()))
+    if (!isChocopyIdentifierBody(*BufPtr))
       IsId = false;
+    readNext();
   }
 
-  if (IsId && isChocopyIdentifierHead(*++Start))
+  const char *End = BufPtr;
+  unsigned Length = End - Start;
+
+  if (IsId && Length > 0 && isChocopyIdentifierHead(*Start))
     Tok.setKind(tok::idstring);
   else
     Tok.setKind(tok::string_literal);
 
-  unsigned Length = BufPtr - Start;
-  SMLoc B = SMLoc::getFromPointer(Start);
-  SMLoc E = SMLoc::getFromPointer(BufPtr);
+  SMLoc B = SMLoc::getFromPointer(QuoteStart);
+  const char *LocEnd = End;
+  if (!isEof() && *BufPtr == '\"')
+    LocEnd = BufPtr + 1;
+  SMLoc E = SMLoc::getFromPointer(LocEnd);
   Tok.setLocation(SMRange(B, E));
   Tok.setLiteralData(Start);
   Tok.setLength(Length);
 
   if (!isEof() && *BufPtr == '\"')
     readNext();
 }
 
 void Lexer::handleToken(Token &Tok, StringRef Str, tok::TokenKind Kind) {
   SymbolInfo &SI = SymbolTable.get(Str, Kind);
   Tok.setKind(SI.getKind());
   SMLoc B = SMLoc::getFromPointer(BufPtr);
   SMLoc E = SMLoc::getFromPointer(BufPtr + SI.getName().size());
 
   Tok.setLocation(SMRange(B, E));
   Tok.setSymbolInfo(&SI);
   Tok.setLength(SI.getName().size());
   BufPtr += Str.size();
 }
-} // namespace chocopy
\ No newline at end of file
+} // namespace chocopy
