diff --git a/chocopy-llvm/lib/Parser/Parser.cpp b/chocopy-llvm/lib/Parser/Parser.cpp
index ad1fa237e816077ed2ccd469304093cfb6b67b94..4faeb055a9326cab62ac1af770bc6a400a5f72f1 100644
--- a/chocopy-llvm/lib/Parser/Parser.cpp
+++ b/chocopy-llvm/lib/Parser/Parser.cpp
@@ -130,50 +130,57 @@ Declaration *Parser::parseDeclaration(bool InClassScope, bool InFunctionScope) {
 ClassDef *Parser::parseClassDef() {
   SMLoc Start = Tok.getLocation().Start;
   consumeToken(); // class
 
   if (!expect(tok::identifier))
     return nullptr;
   SMRange NameLoc = Tok.getLocation();
   SymbolInfo *Name = Tok.getSymbolInfo();
   consumeToken();
 
   if (!expectAndConsume(tok::l_paren))
     return nullptr;
 
   if (!expect(tok::identifier))
     return nullptr;
   SMRange SuperLoc = Tok.getLocation();
   SymbolInfo *Super = Tok.getSymbolInfo();
   consumeToken();
 
   if (!expectAndConsume(tok::r_paren) || !expectAndConsume(tok::colon) ||
       !expectAndConsume(tok::NEWLINE) || !expectAndConsume(tok::INDENT))
     return nullptr;
 
   DeclList Declarations;
   while (Tok.isNot(tok::DEDENT) && Tok.isNot(tok::eof)) {
+    if (Tok.is(tok::kw_pass)) {
+      consumeToken();
+      if (!expectAndConsume(tok::NEWLINE))
+        return nullptr;
+      continue;
+    }
+
     if (Declaration *D = parseDeclaration(/*InClassScope=*/true,
                                           /*InFunctionScope=*/false)) {
       Declarations.push_back(D);
       continue;
     }
 
     // Abort body parse on malformed declaration prefix.
     if (!isDeclStart()) {
       emitUnexpected();
       skipToNextDedent();
       break;
     }
     skipToNextLine();
   }
 
   if (!expectAndConsume(tok::DEDENT))
     return nullptr;
 
   SMLoc End = NameLoc.End;
   if (!Declarations.empty())
     End = Declarations.back()->getLocation().End;
 
   SMRange Loc(Start, End);
   Identifier *ClassId = Context.createIdentifier(NameLoc, Name);
   Identifier *SuperId = Context.createIdentifier(SuperLoc, Super);
@@ -807,55 +814,50 @@ Expr *Parser::parsePrimaryExpr() {
     return parseLiteral();
 
   emitUnexpected();
   return nullptr;
 }
 
 /// type = ID | IDSTRING | '[' type ']'
 TypeAnnotation *Parser::parseType() {
   SMRange Loc = Tok.getLocation();
   switch (Tok.getKind()) {
   case tok::identifier: {
     StringRef Name = Tok.getSymbolInfo()->getName();
     consumeToken();
     return Context.createClassType(Loc, Name);
   }
   case tok::idstring: {
     StringRef Name = Tok.getLiteralData();
     consumeToken();
     return Context.createClassType(Loc, Name);
   }
   case tok::l_square: {
     consumeToken();
     if (TypeAnnotation *T = parseType()) {
       if (expect(tok::r_square)) {
         SMLoc End = Tok.getLocation().End;
-        const char *Ptr = End.getPointer();
-        if (Ptr && *Ptr == ' ')
-          End = SMLoc::getFromPointer(Ptr + 2);
-        else
-          End = SMLoc::getFromPointer(Ptr + 1);
         consumeToken();
         Loc = SMRange(Loc.Start, End);
         return Context.createListType(Loc, T);
       }
     }
     return nullptr;
   }
   default:
     emitUnexpected();
     return nullptr;
   }
 }
 
 Literal *Parser::parseLiteral() {
   SMRange Loc = Tok.getLocation();
 
   if (consumeToken(tok::kw_None)) {
     return Context.createNoneLiteral(Loc);
   } else if (consumeToken(tok::kw_True)) {
     return Context.createBooleanLiteral(Loc, true);
   } else if (consumeToken(tok::kw_False)) {
     return Context.createBooleanLiteral(Loc, false);
   } else if (Tok.is(tok::integer_literal)) {
     llvm::APInt Value(32, Tok.getLiteralData(), 10);
     consumeToken();
